staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager_test.go

```diff
@@ -95,6 +95,63 @@ func TestUpdateOnlyDoesNotTrackManagedFields(t *testing.T) {
 	}
 }
 
+func TestUpdateApplyConflict(t *testing.T) {
+	f := NewTestFieldManager()
+
+	obj := &corev1.Pod{}
+	obj.ObjectMeta.ManagedFields = []metav1.ManagedFieldsEntry{{}}
+
+	_, err := f.Apply(obj, []byte(`{
+		"apiVersion": "apps/v1",
+		"kind": "Deployment",
+		"metadata": {
+			"name": "deployment",
+			"labels": {"app": "nginx"}
+		},
+		"spec": {
+                        "replicas": 3,
+                        "selector": {
+                                "matchLabels": {
+                                         "app": "nginx"
+                                }
+                        },
+                        "template": {
+                                "metadata": {
+                                        "labels": {
+                                                "app": "nginx"
+                                        }
+                                },
+                                "spec": {
+				        "containers": [{
+					        "name":  "nginx",
+					        "image": "nginx:latest"
+				        }]
+                                }
+                        }
+		}
+	}`), "fieldmanager_test", false)
+	if err != nil {
+		t.Fatalf("failed to apply object: %v", err)
+	}
+	newObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
+	newObj.SetAPIVersion("apps/v1")
+	newObj.SetKind("Deployment")
+	newObj.SetName("deployment")
+	_, err = f.Apply(obj, []byte(`{
+		"apiVersion": "apps/v1",
+		"kind": "Deployment",
+		"metadata": {
+			"name": "deployment",
+		},
+		"spec": {
+			"replicas": 101,
+		}
+	}`), "fieldmanager_test", false)
+	if err == nil {
+		// t.Fatalf("Expecting to get conflicts as API error")
```

Comment:
Your test probably fails because because you used apply twice, rather than doing `1. update 2. apply`